core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -4820,42 +4899,39 @@ private VirtualRecordCursorFactory generateSelectVirtualWithSubQuery(QueryModel
                 }
 
                 functions.add(function);
-
+                TableColumnMetadata m = null;
                 if (columnType == ColumnType.SYMBOL) {
                     if (function instanceof SymbolFunction) {
-                        virtualMetadata.add(
-                                new TableColumnMetadata(
-                                        Chars.toString(column.getAlias()),
-                                        function.getType(),
-                                        false,
-                                        0,
-                                        ((SymbolFunction) function).isSymbolTableStatic(),
-                                        function.getMetadata()
-                                )
+                        m = new TableColumnMetadata(
+                                Chars.toString(column.getAlias()),
+                                function.getType(),
+                                false,
+                                0,
+                                ((SymbolFunction) function).isSymbolTableStatic(),
+                                function.getMetadata()
                         );
                     } else if (function instanceof NullConstant) {
-                        virtualMetadata.add(
-                                new TableColumnMetadata(
-                                        Chars.toString(column.getAlias()),
-                                        ColumnType.SYMBOL,
-                                        false,
-                                        0,
-                                        false,
-                                        function.getMetadata()
-                                )
+                        m = new TableColumnMetadata(
+                                Chars.toString(column.getAlias()),
+                                ColumnType.SYMBOL,
+                                false,
+                                0,
+                                false,
+                                function.getMetadata()
                         );
                         // Replace with symbol null constant
                         functions.setQuick(functions.size() - 1, SymbolConstant.NULL);
                     }
                 } else {
-                    virtualMetadata.add(
-                            new TableColumnMetadata(
-                                    Chars.toString(column.getAlias()),
-                                    columnType,
-                                    function.getMetadata()
-                            )
+                    m = new TableColumnMetadata(
+                            Chars.toString(column.getAlias()),
+                            columnType,
+                            function.getMetadata()
                     );
                 }
+                assert m != null;
```

Comment:
Is this assert contradictory to the `if (columnType == ColumnType.SYMBOL)` branch?